src/Features/Core/Portable/ExtractClass/ExtractClassWithDialogCodeAction.cs

```diff
@@ -0,0 +1,264 @@
+                    newType,
+                    symbolMapping,
+                    extractClassOptions.MemberAnalysisResults,
+                    cancellationToken).ConfigureAwait(false);
+
+                return new[] { new ApplyChangesOperation(finalSolution) };
+            }
+            else
+            {
+                // If user click cancel button, options will be null and hit this branch
+                return SpecializedCollections.EmptyEnumerable<CodeActionOperation>();
+            }
+        }
+
+        private async Task<Solution> PullMembersUpAsync(
+            Solution solution,
+            INamedTypeSymbol newType,
+            AnnotatedSymbolMapping symbolMapping,
+            ImmutableArray<ExtractClassMemberAnalysisResult> memberAnalysisResults,
+            CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<(ISymbol member, bool makeAbstract)>.GetInstance(out var pullMembersBuilder);
+            using var _1 = ArrayBuilder<ExtractClassMemberAnalysisResult>.GetInstance(memberAnalysisResults.Length, out var remainingResults);
+            remainingResults.AddRange(memberAnalysisResults);
+
+            // For each document in the symbol mappings, we want to find the annotated nodes
+            // of the members and get the current symbol that represents those after
+            // any changes we made before members get pulled up into the base class.
+            // We only need to worry about symbols that are actually being moved, so we track
+            // the symbols from the member analysis. 
+            foreach (var documentId in symbolMapping.DocumentIds)
+            {
+                if (remainingResults.Count == 0)
+                {
+                    // All symbols have been taken care of
+                    break;
+                }
+
+                var document = solution.GetRequiredDocument(documentId);
+                var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                using var _2 = ArrayBuilder<ExtractClassMemberAnalysisResult>.GetInstance(remainingResults.Count, out var resultsToRemove);
+
+                // Out of the remaining members that we need to move, does this
+                // document contain the definition for that symbol? If so, add it to the builder
+                // and remove it from the symbols we're looking for
+                foreach (var memberAnalysis in remainingResults)
+                {
+                    var annotation = symbolMapping.SymbolToDeclarationAnnotationMap[memberAnalysis.Member];
+
+                    var nodeOrToken = root.GetAnnotatedNodesAndTokens(annotation).SingleOrDefault();
+                    var node = nodeOrToken.IsNode
```

Comment:
maybe be a bit resilient to not finding the node.  it's always a bit of a risk trying to that after edits have already been made.